util/dvsim/Scheduler.py

```diff
@@ -319,28 +317,40 @@ def _dispatch(self, hms):
         def __sum(d):
             return sum([len(d[k]) for k in d])
 
-        num_slots = min(Scheduler.max_parallel - __sum(self._running),
-                        __sum(self._queued))
-        if num_slots <= 0:
+        slots = min(Scheduler.max_parallel - __sum(self._running),
+                    __sum(self._queued))
+        if slots <= 0:
             return
 
+        # Compute how may slots to allocate to each target based on their
+        # weights.
+        sum_weight = 0
+        slots_filled = 0
+        total_weight = sum([
+            self._queued[t][0].weight for t in self._queued if self._queued[t]
+        ])
```

Comment:
Nit: Since `sum` can take a generator, you don't actually need the outer square brackets here.